testsuite/__init__.py

```diff
@@ -0,0 +1,111 @@
+# pylint: disable=missing-docstring
+from flask import Flask, render_template, request
+app = Flask(__name__)  # pylint: disable=invalid-name
+
+
+@app.route("/")
+def index():
+    return render_template('index.html')
+
+
+@app.route('/tests.json', methods=['POST'])
+def tests():
+    from behave.configuration import Configuration
+    from behave.formatter.base import StreamOpener
+    from behave.runner import Runner
+    import io
+
+    output = io.StringIO()
+    output_stream = StreamOpener(stream=output)
+    config = Configuration(outputs=[output_stream], format=['json.pretty'])
+    runner = Runner(config)
+
+    runner.run()
+
+    headers = {
+        'Content-type': 'application/json',
+    }
+
+    return output.getvalue(), 200, headers
+
+
+@app.route('/authorized/')
+def authorized():
+    from flask import jsonify, session, redirect
+    from testsuite.fhir import get_oauth_uris
+    from testsuite import config_reader
+    import requests
+    from testsuite.oauth import RefreshTokenStrategy
+
+    config = config_reader.get_config(session['vendor'])
+    uris = get_oauth_uris(config['api']['url'])
+    strategy = RefreshTokenStrategy(
+        client_id=config['auth']['client_id'],
+        client_secret=config['auth']['client_secret'],
+        redirect_uri=config['auth']['redirect_uri'],
+        urls=uris,
+        refresh_token=config['auth']['refresh_token'],
+        basic=config['auth'].get('basic', False)
+    )
+    strategy.upgrade_authorization_code(request.args.get('code'))
```

Comment:
Just FYI https://tools.ietf.org/html/rfc6749 uses the language "exchange an authorization grant for an access token".